src/Orchard.Web/Modules/Orchard.Layouts/Framework/Drivers/FormsElementDriver.cs

```diff
@@ -32,8 +33,33 @@ public abstract class FormsElementDriver<TElement> : ElementDriver<TElement>, IF
             return Editor(context, formShape);
         }
 
+        /// <summary>
+        /// If we are currently updating the element, then based on the involved forms,
+        /// update the Element's data dictionary with submitted data.
+        /// </summary>
+        protected void UpdateElementData(ElementEditorContext context, IEnumerable<string> formNames, string prefix = "") {
+
+            if (context.Updater != null) {
+
+                // Build forms and look for entires in the Element Data that have the same name.
+                foreach (var formName in formNames) {
+                    var form = _formManager.Build(formName);
+                    FormNodesProcessor.ProcessForm(form, (Action<object>) (shape => {
+                        var name = ((dynamic) shape).Name as string;
+                        if (name != null) {
+                            var value = context.ValueProvider.GetValue(prefix + name);
+                            if (value != null) {
+                                context.Element.Data[name] = value.AttemptedValue;
+                            }
+                        }
+                    }));
+                }
+            }
```

Comment:
@harmony7 after a quick try, indeed posted data are taken into account. Not sure but, as seen in my tests, when you re-edit the element, you don't retreive these data. Here, I needed to add this

            else // here when context.Updater == null
                context.ValueProvider = context.Element.Data.ToValueProvider(CultureInfo.InvariantCulture);

Note: maybe here it's better to use `_cultureAccessor.CurrentCulture` but I would have to change the `FormsElementDriver contructor` that is used in many places...

Best